pkg/signup/service/signup_service.go

```diff
@@ -164,16 +164,27 @@ func EncodeUserID(subject string) string {
 // Signup reactivates the deactivated UserSignup resource or creates a new one with the specified username and userID
 // if doesn't exist yet.
 func (s *ServiceImpl) Signup(ctx *gin.Context) (*toolchainv1alpha1.UserSignup, error) {
-	encodedUserID := EncodeUserID(ctx.GetString(context.SubKey))
+	encodedUserID := EncodeUserIdentifier(ctx.GetString(context.SubKey))
+	var userSignup *toolchainv1alpha1.UserSignup
+	var err error
 	// Retrieve UserSignup resource from the host cluster
-	userSignup, err := s.CRTClient().V1Alpha1().UserSignups().Get(encodedUserID)
+	userSignup, err = s.CRTClient().V1Alpha1().UserSignups().Get(encodedUserID)
 	if err != nil {
 		if apierrors.IsNotFound(err) {
-			// New Signup
-			log.WithValues(map[string]interface{}{"encoded_user_id": encodedUserID}).Info(ctx, "user not found, creating a new one")
-			return s.createUserSignup(ctx)
+			// The UserSignup could not be located by its encoded UserID, attempt to load it using its encoded Username instead
+			encodedUsername := EncodeUserIdentifier(ctx.GetString(context.UsernameKey))
+			userSignup, err = s.CRTClient().V1Alpha1().UserSignups().Get(encodedUsername)
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					// New Signup
+					log.WithValues(map[string]interface{}{"encoded_user_id": encodedUserID}).Info(ctx, "user not found, creating a new one")
+					return s.createUserSignup(ctx)
+				}
+				return nil, err
```

Comment:
Can we add a test for this case? When the client return a non 404 error when trying to get the usersignup.
There are actually two sub-cases: an error when trying to obtain the usersignup by userID and by username. Let's cover at least the username case since this is what we are going to keep after the migration is done.

Reply:
Done